utils/actor-system-error/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![no_std]
+
+#[macro_export]
+macro_rules! actor_system_error {
+    (
+        $(#[$($meta:meta)*])?
+        $vis:vis type $name:ident = ActorSystemError<$actor_err:ident, $system_err:ident>;
+    ) => {
+        $(#[$($meta)*])?
+        $vis type $name = $crate::ActorSystemError<$actor_err, $system_err>;
+
+        impl From<$actor_err> for $crate::ActorSystemError<$actor_err, $system_err> {
+            fn from(err: $actor_err) -> Self {
+                Self::Actor(err)
+            }
+        }
+
+        impl From<$system_err> for $crate::ActorSystemError<$actor_err, $system_err> {
+            fn from(err: $system_err) -> Self {
+                Self::System(err)
+            }
+        }
+    };
+}
+
+#[derive(Debug, Eq, PartialEq, derive_more::Display)]
+pub enum ActorSystemError<A, S> {
```

Comment:
mb apply `std::error::Error` to this as well?

Reply:
We mainly work in non-std environment so It's impossible to apply